api_app/services/aad_authentication.py

```diff
@@ -278,21 +279,31 @@ def _get_users_inc_groups_from_response(self, users_graph_data, roles_graph_data
                 user_name = user_data["body"]["displayName"]
 
                 if "users" in user_data["body"]["@odata.context"]:
-                    user_email = user_data["body"]["mail"]
+                    user_email = user_data["body"]["userPrincipalName"]
```

Comment:
The UPN is often different to the user's email? Why is this happening?

Reply:
fixed (renamed)